backend/app/processing/kb.py

```
@@ -265,37 +265,54 @@ def search_kb(chatbot_id: str, query: str, k: int = 5) -> List[str]:
+        for root, _, files in os.walk(directory):
+            for fname in files:
+                if is_ignored_file(fname):
+                    continue
+                full_path = os.path.join(root, fname)
+                if is_supported_file(full_path):
+                    collected_files.append(full_path)
+    return collected_files
+
+
+def is_ignored_file(fname: str) -> bool:
+    return fname.startswith('.') or fname.startswith('~$')
```

Ruff (0.8.2)

Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)